src/System.Diagnostics.Process/src/Resources/Strings.resx

```diff
@@ -187,6 +187,9 @@
   <data name="ProcessorAffinityNotSupported" xml:space="preserve">
     <value>Processor affinity for processes or threads is not supported.</value>
   </data>
+  <data name="ProcessStartIdentityNotSupported" xml:space="preserve">
+    <value>Process start with identity is not supported on this platform.</value>
```

Comment:
Better, thanks. Though technically it's still a bit off: this will throw even if you try to specify the credentials of the same (current) user. Maybe that's too nit picky. You could address it by changing "different" to "specified" or something like that.